var/spack/repos/builtin/packages/icu4c/package.py

```diff
@@ -42,7 +42,8 @@ def url_for_version(self, version):
 
     def install(self, spec, prefix):
         with working_dir('source'):
-            configure('--prefix={0}'.format(prefix))
+            configure('--prefix={0}'.format(prefix),
+                      '--enable-rpath')
```

Comment:
WIthout this, the ID on OS X is just a relative path rather than a full path. When Qt links it, it uses the ID and tries to open `libicu4i18n.dylib` relative to the directory containing `QtCore.framework` which doesn't work.